src/transaction_processor.cpp

```diff
@@ -86,9 +108,12 @@ void TransactionProcessorImpl::Register() {
             TpRegisterRequest request;
             request.set_family(handler.second->transaction_family_name());
             request.set_version(version);
+            request.set_protocol_version(
+                static_cast<unsigned int>(this->highest_sdk_feature_requested));
```

Comment:
These static_casts seem unfortunate.  Is there anyway we can avoid that?

Reply:
The validator to TP interface is setup as an unsigned int, however for the developers - there's option to choose the version they want from the given enum. I am unable to think of another approach. Are you suggesting to add a custom converter function or use constant instead of enum?

Reply:
I'd go with a converter as opposed to constants.  The converter would let you avoid the cast as well as fall back to some sensible default if that is desirable.

Reply:
Attempted the custom converter with a switch.

Reply:
Looks good to me